src/VS.Mefx/Commands/PartNode.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace VS.Mefx.Commands
+{
+    using System.Collections.Generic;
+    using Microsoft.VisualStudio.Composition;
+
+    /// <summary>
+    /// Node object to represent parts when performing rejection tracing.
+    /// </summary>
+    internal class PartNode
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PartNode"/> class.
+        /// </summary>
+        /// <param name="definition">The definition of the part associated with the node.</param>
+        /// <param name="message">Initialize rejection message in error stack.</param>
+        /// <param name="currLevel">The depth of the part in the stack.</param>
+        public PartNode(ComposablePartDefinition definition, string message, int currLevel)
+        {
+            this.Part = definition;
+            this.IsWhiteListed = false;
+            this.VerboseMessages = new HashSet<string> { message };
+            this.ImportRejects = new HashSet<PartEdge>();
+            this.RejectsCaused = new HashSet<PartEdge>();
+            this.Level = currLevel;
+            this.IsWhiteListed = false;
+
+            this.ExportingContracts = new List<string>();
+            foreach (var export in this.Part.ExportDefinitions)
+            {
+                if (export.Key == null)
+                {
+                    continue;
+                }
+
+                this.ExportingContracts.Add(export.Value.ContractName);
+            }
+        }
+
+        /// <summary>
+        /// Gets the definition associated with the current part.
+        /// </summary>
+        public ComposablePartDefinition Part { get; private set; } // Represents the part associated with the current node
```

Comment:
Can we drop the `//` code comment since we already have a `/// <summary>` comment?